internal/operators/api/api.go

```diff
@@ -0,0 +1,58 @@
+package api
+
+import (
+	"context"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+)
+
+type ValidationStatus string
+
+const (
+	Success ValidationStatus = "success"
+	Failure ValidationStatus = "failure"
+	Pending ValidationStatus = "pending"
+)
+
+// ValidationResult hold result of operator validation
+type ValidationResult struct {
+	// ValidationId is an id of the validation
+	ValidationId string
+	// Status specifies the status of the validation: success, failure or pending
+	Status ValidationStatus
+	// Reasons hold list of reasons of a validation failure
+	Reasons []string
+}
+
+// Manifests hold generated manifests
+type Manifests struct {
+	// Files maps file name to file's content
+	Files map[string]string
+}
+
+// Operator provides generic API of an OLM operator installation plugin
+type Operator interface {
```

Comment:
If `IsEnabled` is needed anywhere, it might be promoted to exported method of the `operators.Manager`, as the logic is generic and more cluster- than operator-oriented, but so far I haven't found any use for it in the code.

I haven't found use for `getVersion` in the current code.

Good point about `GetDiskRequirements` - I will add API method for reporting minimal required number of disks for master and worker nodes - like for the CPU and RAM.

Reply:
Added `GetDisksRequirementForWorker` and `GetDisksRequirementForMaster` to the API.